### DISCUSSION TITLE
Tiled plots removing oldest plots when window resizing

### DISCUSSION THREAD

**Discussion by matthew-hennefarth** [alandefreitas/matplotplusplus#160] [Q&A] [answered]
I am having issues plotting, say 4 plots in a square grid
![original](https://user-images.githubusercontent.com/27997339/121432801-fa9a6380-c92f-11eb-932f-538fc57e7818.png)

If I resize the gnuplot window, it deletes plots 0, 1, and 2, but keeps 3. This also occurs if I press the gnuplot 'refresh' button.
![resized](https://user-images.githubusercontent.com/27997339/121432866-10a82400-c930-11eb-9ff1-0440399158b4.png)

I am unsure if this is an issue with gnuplot or some way matplotplusplus is drawing. Or it could be that I am missing some key word in my plotting. Here is the code I am using with some comments to understand the variables.

```c++
    /* current_ax is defined */
    /* rotatedElectricFields is a std::array<std::vector<double>, 4> */
    /* plotX_, plotY_, plotZ_, plotM_ are member functions that return bool */
    
    size_t plot_index = 0; /* tracks plot index (ie 0, 1, 2, 3) */

    const auto plot = [&rotatedElectricFields, &plot_index, &current_ax](size_t index) {
      constexpr std::array<const char*, 4> titles{"X", "Y", "Z", "Magnitude"};
      current_ax = matplot::nexttile(plot_index);
      matplot::hold(current_ax, matplot::on);
      matplot::plot(current_ax, rotatedElectricFields.at(index));
      current_ax->xlabel("Frame");
      current_ax->ylabel("Magnitude (V/Ang)");
      current_ax->title(titles.at(index));
      ++plot_index;
      matplot::hold(current_ax, matplot::off);
    };

    if (plotX_()) {
      plot(0);
    }
    if (plotY_()) {
      plot(1);
    }
    if (plotZ_()) {
      plot(2);
    }
    if (plotM_()) {
      plot(3);
    }

    matplot::show();
    return;
```

If this is an error with matplot, we can change this to an issue and I can try and work on a fix to it. Thanks!

**Comment by alandefreitas** [comment, marked as answer]
This happens with Gnuplot all the time with multi plots. One way to get around that is to resize the window programmatically to the size you want before plotting. 

You can also build it with `static constexpr bool trace_commands = true;` in `gnuplot.h` to see the Gnuplot output in the console and then paste that in gnuplot to check if you get the same output. I can't say for sure but I imagine the result would be the same.

**Reply by matthew-hennefarth** [reply]
That makes sense that it is a gnuplot issue. Here are the output plotting commands:
[output.txt](https://github.com/alandefreitas/matplotplusplus/files/6627344/output.txt)

Interestingly, when I run this as `gnuplot --persist output.txt` the window resizes normally, so I wonder what is causing the discrepency between the two.

**Reply by matthew-hennefarth** [reply]
ah, found the gnuplot [explanation](http://www.gnuplot.info/faq/#x1-780008.5) It is probably because gnuplot is using this file to replot, whereas in matplot++ it is using the pipe.

Since it seems I am using qt on my mac, I just have to go to the toolbar and make sure the replot on resize is unchecked:
<img width="340" alt="Screen Shot 2021-06-09 at 16 24 06" src="https://user-images.githubusercontent.com/27997339/121441719-2e30ba00-c93f-11eb-8386-ee2295863e66.png">

**Reply by alandefreitas** [reply]
That's very interesting. There's probably a way to set this option from the pipe before plotting.

**Reply by matthew-hennefarth** [reply]
They said for x11 terminals you can with `set term x11 noreplotonresize` but for qt and wxt it is only available through the toolbox in the window. It may be good to use `set term x11 noreplotonresize` if the x11 terminal is used and multiplot is used? Or at least make it an option. 

Though, I will say, it seems to persist across multiple "gnuplot sessions" with matplot++. That is, if I run the program once, and turn off replot on resize, finish the program, and then re-run the program, it will already have the replot on resize turned off.

**Reply by alandefreitas** [reply]
Yes. That seems reasonable. Matplot++ already sets some options according to the terminal being used. It's reasonably safe to assume people want their plots to work after resizing because not reporting destroys the whole thing. It's reasonably safe to assume people want everything to work all the time actually 😆.
